src/NuGetGallery.Core/GitHub/GitHubUsageConfiguration.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGetGallery.GitHub
+{
+    public class GitHubUsageConfiguration : IGitHubUsageConfiguration
+    {
+        public GitHubUsageConfiguration(IReadOnlyList<RepositoryInformation> repositories)
+        {
+            if (repositories == null)
+            {
+                throw new ArgumentNullException(nameof(repositories) + " is null!");
+            }
+
+            NuGetPackagesGitHubDependencies = repositories.Any()
+                ? GetNuGetPackagesDependents(repositories)
+                : new Dictionary<string, NuGetPackageGitHubInformation>();
+        }
+
+        private Dictionary<string, NuGetPackageGitHubInformation> NuGetPackagesGitHubDependencies { get; }
+
+        public NuGetPackageGitHubInformation GetPackageInformation(string packageId)
+        {
+            if (null == packageId)
+            {
+                throw new ArgumentException(string.Format("{0} cannot be null!", nameof(packageId)));
+            }
+
+            if (NuGetPackagesGitHubDependencies.TryGetValue(packageId, out var value))
+            {
+                return value;
+            }
+
+            return new NuGetPackageGitHubInformation();
+        }
+
+        private static Dictionary<string, NuGetPackageGitHubInformation> GetNuGetPackagesDependents(IReadOnlyList<RepositoryInformation> repositories)
+        {
+            var dependentsPerPackage = new Dictionary<string, List<RepositoryInformation>>(StringComparer.InvariantCultureIgnoreCase);
```

Comment:
It may be easier to sort `repositories` list first, instead of sorting individual entries from `dependentsPerPackage` at the end. This may be a bad idea

Reply:
Did some benchmarking (not really scientific tho), and it wasn't faster, also to get the total dependents count, I would have to have another Dictionary<string, int> that keeps track of that, which slows down a bit the execution.

Reply:
Great, thanks for looking into it!